src/uct/api/uct.h

```diff
@@ -840,7 +968,8 @@ struct uct_ep_params {
     uint64_t                          field_mask;
 
     /**
-     * Interface to create the endpoint on. This is a mandatory field.
+     * Interface to create the endpoint on.
+     * Either @a iface or @a cm field must be initialized but not both.
```

Comment:
why ? does it mean we need a union of the two ?

Reply:
@shamisp opening an endpoint on an iface is basically selecting the previous API while opening on a CM is using this new one. Having both of them in this struct would enable the user to select between them (same as selecting other fields here for usage).